src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -118,19 +116,23 @@ public boolean Disconnect(String peerId){
     public boolean  DisconnectIncomingConnections(){
 
         boolean ret = false;
+
         if (mServerSocketList != null) {
-            for(int i = 0; i < mServerSocketList.size(); i++) {
-                BtToServerSocket tmp = mServerSocketList.get(i);
-                if (tmp != null) {
-                    print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
-                    tmp.Stop();
-                    tmp = null;
-                    mServerSocketList.set(i, tmp);
-                    ret = true;
+            synchronized(mServerSocketList) {
+
+                for (int i = 0; i < mServerSocketList.size(); i++) {
```

Comment:
@DrJukka Note that when you switch to CopyOnWriteArrayList or equivalent you will want to use methods like elements() to get a consistent set of contents to iterate over.